### PR TITLE
Multi-database support + added ENUM and SET for MySQL only

### PR DESCRIPTION
**Pull request by bartekrylko**
Added support for multi-databases migrations.
Added also support for ENUM and SET MySQL types. Values should be defined in 'limit' field.
More about using multi-db can be found here: https://github.com/DatingVIP/phinx/tree/0.4.x-dev

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,4 @@ phpunit.xml
 
 # sqlite test database
 phinx_testing.sqlite3
+/nbproject/
```
Comment (shadowhand) [inline .gitignore]: These kinds of user-specific ignores should be done via [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) not at the repository level.

File: README.md [hunk 2]
```diff
@@ -31,6 +31,14 @@ Phinx natively supports the following database adapters:
 * SQLite
 * Microsoft SQL Server
 
+#### Features added in this fork
```
Comment (shadowhand) [inline README.md]: These notes should be added to the documentation files, not the README.

File: src/Phinx/Config/Config.php [hunk 3]
```diff
@@ -143,7 +143,7 @@ public function getEnvironments()
     /**
      * {@inheritdoc}
      */
-    public function getEnvironment($name)
+    public function getEnvironment($name, $database = null)
```
Comment (shadowhand) [inline src/Phinx/Config/Config.php]: These changes for multi-database should be a separate PR from adding new column types, as the two are completely unrelated.

**Comment by shadowhand** [comment]
The file mode changes `100644 → 100755` should be reverted... Only executable files should have the executable bit set.

File: src/Phinx/Db/Adapter/AdapterWrapper.php [hunk 4]
```diff
@@ -227,7 +227,7 @@ public function isValidColumnType(Column $column)
      */
     public function hasTransactions()
     {
-        return $this->getAdapter()->hasTransaction();
+        return $this->getAdapter()->hasTransactions();
```
Comment (shadowhand) [inline src/Phinx/Db/Adapter/AdapterWrapper.php]: Good catch, but this should have been a separate PR. I'll make the change to master.

File: src/Phinx/Db/Table/Column.php [hunk 5]
```diff
@@ -151,7 +151,7 @@ public function getType()
     /**
      * Sets the column limit.
      *
-     * @param integer $limit
+     * @param mixed $limit integer|string - can be string for SET and ENUM
```
Comment (shadowhand) [inline src/Phinx/Db/Table/Column.php]: This is pretty weird to me, I don't think using `limit` for ENUM options makes sense... and shouldn't ENUM options be an array, not a string?

File: tests/Phinx/Db/Adapter/MysqlAdapterTest.php [hunk 6]
```diff
@@ -871,4 +871,32 @@ public function testHasColumnReservedName()
         $this->assertTrue($tableQuoted->hasColumn('value'));
 
     }
+
+    public function testGetColumnTypes()
```
Comment (shadowhand) [inline tests/Phinx/Db/Adapter/MysqlAdapterTest.php]: If the intent of this test is simply to make sure that `enum` and `set` are in the column types, that should be the only thing tested. As currently written, this test is fragile because adding new base types will cause it to fail.

File: src/Phinx/Db/Table/Column.php (hunk 5 above)
Comment (bartekrylko) [inline src/Phinx/Db/Table/Column.php]: What do you suggest here? New column option called 'values' for example?
That option could take both, string and array.

Comment (shadowhand) [inline src/Phinx/Db/Table/Column.php]: Yes, a new column option. Calling it "values" makes sense to me.

**Comment by bartekrylko** [comment]
Splitted into two separate: #464 #464 with suggested changes.
